src/services/geometry/richgeo/ReadoutGeo.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2023, Christopher Dilks
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+
+// helper functions for RICH readout
+
+#pragma once
+
+#include <string>
+#include <fmt/format.h>
+#include <functional>
+#include <TRandomGen.h>
+
+// DD4Hep
+#include <DD4hep/Detector.h>
+#include <DD4hep/DD4hepUnits.h>
+#include <DDSegmentation/BitFieldCoder.h>
+
+// local
+#include "RichGeo.h"
+
+namespace richgeo {
+  class ReadoutGeo {
+    public:
+
+      // constructor
+      ReadoutGeo(std::string detName_, dd4hep::Detector *det_, bool verbose_=false);
+      ~ReadoutGeo() {}
+
+      // random number generators
+      TRandomMixMax m_random;
```

Comment:
- Can this be moved to `private`?
- Can this take a `seed`, perhaps the one from `PhotoMultiplierHitDigiConfig`?
- Thread unsafe, but that's a [separate known issue](https://github.com/eic/EICrecon/issues/539)